projects/plugins/jetpack/modules/sharedaddy/sharing-sources.php

```diff
@@ -201,19 +213,53 @@ public function get_link( $url, $text, $title, $query = '', $id = false ) {
 		 *
 		 * @param string $text Sharing service text.
 		 * @param object $this Sharing service properties.
-		 * @param int|false $id Sharing ID.
+		 * @param string|false $id Sharing ID.
 		 * @param array $args Array of sharing service options.
 		 */
 		$text = apply_filters( 'jetpack_sharing_display_text', $text, $this, $id, $args );
 
+		/**
+		 * Filter the sharing data attributes.
+		 *
+		 * @module sharedaddy
+		 *
+		 * @since 10.10.0
+		 *
+		 * @param array $data_attributes Attributes supplied from the sharing source.
+		 *                               Note that 'data-' will be prepended to all keys.
+		 * @param Sharing_Source $this Sharing source instance.
+		 * @param string|false $id Sharing ID.
+		 * @param array $args Array of sharing service options.
+		 */
+		$data_attributes = apply_filters( 'jetpack_sharing_data_attributes', $data_attributes, $this, $id, $args );
```

Comment:
Should we cast this as an array, to avoid any potential warnings downer if `$data_attributes` is not an iterable?